File: server/makc2022--nodejs--nestjs/src/layer3/nosql-mongo/sample/entities/dummy-main-dummy-many-to-many/dummy-main-dummy-many-to-many-entity.loader.ts
/** Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License. */

import { nameof } from '../../../../../layer1/common/common-functions';
import { EntityLoader } from '../../../../../layer1/entity/entity-loader';
import { DummyMainDummyManyToManyEntityObject } from './dummy-main-dummy-many-to-many-entity.object';

/** Загрузчик сущности "DummyMainDummyManyToMany". */
export class DummyMainDummyManyToManyEntityLoader extends EntityLoader<DummyMainDummyManyToManyEntityObject> {
  /** @inheritdoc */
  constructor(entityObject?: DummyMainDummyManyToManyEntityObject) {
    super(entityObject ?? ({} as DummyMainDummyManyToManyEntityObject));
  }

  /** @inheritdoc */
  override load(
    entityObject: DummyMainDummyManyToManyEntityObject,
    loadableProperties?: Set<string>
  ): Set<string> {
    const result = super.load(entityObject, loadableProperties);

    if (result.has(nameof<DummyMainDummyManyToManyEntityObject>('idOfDummyMainEntity'))) {
      this.entityObject.idOfDummyMainEntity = entityObject.idOfDummyMainEntity;
    }

    if (result.has(nameof<DummyMainDummyManyToManyEntityObject>('idOfDummyManyToManyEntity'))) {
      this.entityObject.idOfDummyManyToManyEntity = entityObject.idOfDummyManyToManyEntity;
    }

    return result;
  }

  /** @inheritdoc */
  protected override createAllPropertiesToLoad(): Set<string> {
    return new Set<string>([
      nameof<DummyMainDummyManyToManyEntityObject>('idOfDummyMainEntity'),
      nameof<DummyMainDummyManyToManyEntityObject>('idOfDummyManyToManyEntity'),
    ]);
  }
}

File: server/makc2022--dotnet/Makc2022.Layer3.Sql.Sample/Entities/DummyMainDummyManyToMany/DummyMainDummyManyToManyEntityLoader.cs
// Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

using Makc2022.Layer3.Sql.Sample.Entity;

namespace Makc2022.Layer3.Sql.Sample.Entities.DummyMainDummyManyToMany
{
    /// <summary>
    /// Загрузчик сущности "DummyMainDummyManyToMany".
    /// </summary>
    public class DummyMainDummyManyToManyEntityLoader : EntityLoader<DummyMainDummyManyToManyEntityObject>
    {
        #region Constructors

        /// <inheritdoc/>
        public DummyMainDummyManyToManyEntityLoader(DummyMainDummyManyToManyEntityObject? entityObject = null)
            : base(entityObject ?? new DummyMainDummyManyToManyEntityObject())
        {
        }

        #endregion Constructors

        #region Public methods

        /// <inheritdoc/>
        public sealed override HashSet<string> Load(DummyMainDummyManyToManyEntityObject entityObject, HashSet<string>? loadableProperties = null)
        {
            var result = base.Load(entityObject, loadableProperties);

            if (result.Contains(nameof(EntityObject.IdOfDummyMainEntity)))
            {
                EntityObject.IdOfDummyMainEntity = entityObject.IdOfDummyMainEntity;
            }

            if (result.Contains(nameof(EntityObject.IdOfDummyManyToManyEntity)))
            {
                EntityObject.IdOfDummyManyToManyEntity = entityObject.IdOfDummyManyToManyEntity;
            }

            return result;
        }

        #endregion Public methods

        #region Protected methods

        /// <inheritdoc/>
        protected override HashSet<string> CreateAllPropertiesToLoad()
        {
            return new HashSet<string>
            {
                nameof(EntityObject.IdOfDummyMainEntity),
                nameof(EntityObject.IdOfDummyManyToManyEntity)
            };
        }

        #endregion Protected methods
    }
}
